pypeit/flatfield.py

```diff
@@ -384,11 +518,13 @@ def show(self, wcs_match=True):
             wcs_match (:obj:`bool`, optional):
                 Match the WCS of the flat-field images
         """
-        # Get the slits
-        show_flats(self.mspixelflat, self.msillumflat, self.rawflatimg.image, self.flat_model,
-                   wcs_match=wcs_match, slits=self.slits)
+        # Prepare the images to show, their names and their cuts
+        image_list = zip([self.mspixelflat, self.msillumflat, self.rawflatimg.image, self.flat_model],
+                         ['pixelflat', 'spat_illum', 'raw', 'model', 'spec_illum'],
+                         [(0.9, 1.1), (0.9, 1.1), None, None, (0.8, 1.2)])
+        show_flats(image_list, wcs_match=wcs_match, slits=self.slits)
 
-    def fit(self, debug=False):
+    def fit(self, illumflat=False, debug=False):
```

Comment:
The default behavior of this routine should be to generate both pixel and illumination flats. There should be an option to short circuit that called spat_illum_only or something like that. This is how your codes work above, but you have switched the convention here and set the default to be not creating the illumflat. I think you should use the same spat_illum_only here, and the default should spat_illum_only=False. The docs should be changed to state that if True, only the illumination flat will be generated and the code will exit.